docs/src/app/(docs)/react/utils/merge-props/page.mdx

```diff
@@ -0,0 +1,57 @@
+# mergeProps
+
+<Subtitle>A utility to merge multiple sets of React props.</Subtitle>
+<Meta
+  name="description"
+  content="A utility to merge multiple sets of React props, handling event handlers, className, and style props intelligently."
+/>
+
+`mergeProps` helps you combine multiple prop objects (for example, internal props + user props) into a single set of props you can spread onto an element.
+It behaves like `Object.assign` (rightmost wins) with a few special cases, so common React patterns work as expected.
+
+## How merging works
+
+- For most keys (everything except `className`, `style`, and event handlers), the value from the rightmost object wins:
+  ```ts
+  mergeProps({ id: 'a', dir: 'ltr' }, { id: 'b' }); // returns { id: 'b', dir: 'ltr' }
+  ```
+  Note that this includes refs.
```

Comment:
I don't think this is the behavior most people would expect without diving deeper into why we don't merge the refs.
So I think it's worth being extra-explicit here and add a bullet point specifically for the `ref` 👍